worlds/ff1/Client.py

```diff
@@ -0,0 +1,298 @@
+            elif current_item_name in armor:
+                self.armor_queue.appendleft(current_item_id - 0x143)
+            await self.write_sram(ctx, items_obtained, items_received_count + 1)
+
+    async def process_weapons_queue(self, ctx):
+        empty_slots = deque()
+        char1_slots = await self.read_sram_values(ctx, weapons_arrays_starts[0], 4)
+        char2_slots = await self.read_sram_values(ctx, weapons_arrays_starts[1], 4)
+        char3_slots = await self.read_sram_values(ctx, weapons_arrays_starts[2], 4)
+        char4_slots = await self.read_sram_values(ctx, weapons_arrays_starts[3], 4)
+        for i, slot in enumerate(char1_slots):
+            if slot == 0:
+                empty_slots.appendleft(weapons_arrays_starts[0] + i)
+        for i, slot in enumerate(char2_slots):
+            if slot == 0:
+                empty_slots.appendleft(weapons_arrays_starts[1] + i)
+        for i, slot in enumerate(char3_slots):
+            if slot == 0:
+                empty_slots.appendleft(weapons_arrays_starts[2] + i)
+        for i, slot in enumerate(char4_slots):
+            if slot == 0:
+                empty_slots.appendleft(weapons_arrays_starts[3] + i)
+        while len(empty_slots) > 0 and len(self.weapons_queue) > 0:
+            current_slot = empty_slots.pop()
+            current_weapon = self.weapons_queue.pop()
+            await self.write_sram(ctx, current_slot, current_weapon)
+
+    async def process_armor_queue(self, ctx):
+        empty_slots = deque()
+        char1_slots = await self.read_sram_values(ctx, armors_arrays_starts[0], 4)
+        char2_slots = await self.read_sram_values(ctx, armors_arrays_starts[1], 4)
+        char3_slots = await self.read_sram_values(ctx, armors_arrays_starts[2], 4)
+        char4_slots = await self.read_sram_values(ctx, armors_arrays_starts[3], 4)
+        for i, slot in enumerate(char1_slots):
+            if slot == 0:
+                empty_slots.appendleft(armors_arrays_starts[0] + i)
+        for i, slot in enumerate(char2_slots):
+            if slot == 0:
+                empty_slots.appendleft(armors_arrays_starts[1] + i)
+        for i, slot in enumerate(char3_slots):
+            if slot == 0:
+                empty_slots.appendleft(armors_arrays_starts[2] + i)
+        for i, slot in enumerate(char4_slots):
+            if slot == 0:
+                empty_slots.appendleft(armors_arrays_starts[3] + i)
+        while len(empty_slots) > 0 and len(self.armor_queue) > 0:
+            current_slot = empty_slots.pop()
+            current_armor = self.armor_queue.pop()
+            await self.write_sram(ctx, current_slot, current_armor)
+
+    async def read_ram_values(self, ctx, location, size):
+        return (await bizhawk.read(ctx.bizhawk_ctx, [(location, size, self.wram)]))[0]
+
+    async def read_ram_value(self, ctx, location):
+        value = ((await bizhawk.read(ctx.bizhawk_ctx, [(location, 1, self.wram)]))[0])
+        return int.from_bytes(value)
```

Comment:
All of these `int.from_bytes` will crash in Python 3.10
The endianness needs to be specified.
(The default value was added in Python 3.11)
```suggestion
        return int.from_bytes(value, "little")
```